spec/controllers/user_tasks_controller_spec.rb

```diff
@@ -0,0 +1,83 @@
+require 'rails_helper'
+
+RSpec.describe UserTasksController, type: :controller do
+  let(:user) { FactoryGirl.create(:user) }
+  let(:task) { FactoryGirl.create(:task) }
+
+  context "when user is logged" do
+    before do
+      sign_in(user)
+    end
+
+    context "with proper params" do
+      context "#assign" do
+        before do
+          post :assign, params: { project_id: task.project.id, id: task.id }
+        end
+
+        it "assigns task to the user" do
+          expect(assigns(:user_task)).to eql(UserTask.last)
+        end
+
+        it "returns flash success message" do |variable|
+          expect(flash[:notice]).to include("Successfully assigned!")
```

Comment:
remember to change compared string here if you decide to change notice message in the controller